packages/app/obojobo-repository/shared/components/dashboard.jsx

```diff
@@ -65,62 +65,113 @@ const renderModalDialog = props => {
 	)
 }
 
-const renderModules = modules => {
-	return modules.map(draft => <Module key={draft.draftId} hasMenu={true} {...draft}></Module>)
-}
+class Dashboard extends React.Component {
+	constructor(props) {
+		super(props)
 
-const Dashboard = props => {
-	return (
-		<span id="dashboard-root">
-			<RepositoryNav
-				userId={props.currentUser.id}
-				avatarUrl={props.currentUser.avatarUrl}
-				displayName={`${props.currentUser.firstName} ${props.currentUser.lastName}`}
-				noticeCount={0}
-			/>
-			<RepositoryBanner title={props.title} className="default-bg" facts={props.facts} />
-			<div className="repository--section-wrapper">
-				<section className="repository--main-content">
-					<div className="repository--main-content--control-bar">
-						<MultiButton title="New Module">
-							<Button
-								onClick={() => {
-									props.createNewModule(false)
-								}}
-							>
-								New Module
-							</Button>
-							<Button
-								onClick={() => {
-									props.createNewModule(true)
-								}}
-							>
-								New Tutorial
-							</Button>
-						</MultiButton>
-						<Search
-							value={props.moduleSearchString}
-							placeholder="Filter..."
-							onChange={props.filterModules}
-						/>
-					</div>
-					<div className="repository--main-content--title">
-						<span>My Modules</span>
-					</div>
-					<div className="repository--item-list--collection">
-						<div className="repository--item-list--collection--item-wrapper">
-							<div className="repository--item-list--row">
-								<div className="repository--item-list--collection--item--multi-wrapper">
-									{renderModules(props.filteredModules ? props.filteredModules : props.myModules)}
+		const selectedSort = props.selectedSort
+
+		this.state = {
+			selectedSort,
+			sortOptions: ['Newest', 'Alphabetical', 'Last updated']
+		}
+	}
+
+	updateSort(event) {
+		const selectedSort = event.target.value
+		const expires = new Date()
+
+		this.setState({ selectedSort })
+
+		expires.setFullYear(expires.getFullYear() + 1)
+		document.cookie = `selectedSort=${selectedSort}; expires=${expires.toUTCString()}`
+	}
+
+	renderModules(modules) {
+		switch (this.state.selectedSort) {
+			case 'Alphabetical':
+				modules.sort((a, b) => a.title.localeCompare(b.title))
+				break
+			case 'Newest':
+				modules.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt))
+				break
+			case 'Last updated':
+				modules.sort((a, b) => new Date(b.updatedAt) - new Date(a.updatedAt))
+				break
```

Comment:
Not a required change, but I'd try pulling the call to `modules.sort` out of the switch, instead store the sorting function in a variable like `sortFn` then call `modules.sort(sortFn)` after the switch.

Possible reasons why: it's nice to limit what happens in the logical forks in the code to just what's different. And it makes it clear that all paths lead to sorting.

That said, like any polishing step - try it out and if it's not any better don't do it.